tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+                            .resolve().getType());
+        } catch (UnsolvedSymbolException e) {
+            logger.warn("Failed to evaluate instanceof within method. Expression: \"type1Var instanceof {}\". Method: {}", type2, syntheticMethodBody.getParentNode().get());
+            return false;
+        }
+    }
+
+    private static ResolvedType tryToGetResolvedType(String type2) {
+        ResolvedType resolvedType2;
+        try {
+            resolvedType2 = getResolvedType(type2);
+        } catch (UnsolvedSymbolException e) {
+            return null;
+        }
+        return resolvedType2;
+    }
+
+    // TODO: The "not the other way around" is true only in a certain sense.  Given variables `b`
+    // and `B` of type boolean and Boolean, both "b = B" and "B = b" will compile, because Java
+    // performs automatic boxing and unboxing.  Please clarify the comment.
+    /**
+     * This method is different from {@link #isInstanceOf} in that it can be used to compare
+     * primitive types and primitive wrapper types. For instance, a boolean is assignable to a Boolean,
+     * but not the other way around. Note that this method takes as input pairs of package and class
+     * name, instead of fully qualified types.
+     * @param type1 pair with &lt;package, class&gt;, e.g., &lt;"java.util", "List"&gt;
+     * @param type2 pair with &lt;package, class&gt;, e.g., &lt;"java.lang", "Object"&gt;
+     * @param oracleDatapoint may be null. If not null, it is used to check if some type is generic.
+     * @return true if a variable of type1 can be assigned to a variable of type2, false otherwise
+     */
+    public static boolean isType1AssignableToType2(Pair<String, String> type1, Pair<String, String> type2, OracleDatapoint oracleDatapoint) {
+        // Cases to consider:
+        // 1) left and right types are the same
+        // 2) left type is null and right type is not primitive
+        // 3) left type is primitive and right type is wrapper
+        // 4) both types are numeric and left type is assignable to right type
+        // 5) left type is instance of right type (both types are non-primitive)
+        return
+                type1.equals(type2) ||
+                (type1.equals(JavaTypes.NULL) && !JavaTypes.PRIMITIVES.contains(type2)) ||
+                (JavaTypes.PRIMITIVES.contains(type1) && JavaTypes.PRIMITIVES_TO_WRAPPERS.get(type1).equals(type2)) ||
+                (JavaTypes.NUMBERS.contains(type1) && JavaTypes.NUMBERS.contains(type2) && isAssignableToNumeric(type1, type2)) ||
+                isInstanceOf(fullyQualifiedClassName(type1), fullyQualifiedClassName(type2), oracleDatapoint);
+    }
+
+    public static TypeDeclaration<?> getClassOrInterface(CompilationUnit cu, String name) {
+        try {
+            return cu.getLocalDeclarationFromClassname(name).get(0);
+        } catch (NoSuchElementException|IndexOutOfBoundsException ignored) {}
```

Comment:
Please don't use exceptions for control flow.  It is harder to read and understand.  Instead, do 
```java
someVar = cu.getLocalDeclarationFromClassname(name);
if (someCondition(someVar)) {
  return someVar;
}
...
```

Reply:
In this particular case (the first one), `getLocalDeclarationFromClassname` might throw a `NoSuchElementException`, so we need the try-catch block. In the remaining there's no need, so I've replaced them as you suggested.

See https://github.com/AML14/tratto/pull/41/commits/35200a58e55bc699bf24805cf5a519348dffe1f2.